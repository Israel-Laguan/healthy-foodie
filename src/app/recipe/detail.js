import React, { useEffect, useCallback } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import {
  Row,
  Col,
  Badge,
  Breadcrumb,
  BreadcrumbItem,
  Jumbotron,
} from "reactstrap";

export function Recipe({ selectedRecipe, ...props }) {
  const toHome = useCallback(() => {
    props.history.push(`../home`);
  }, [props.history]);

  useEffect(() => {
    if (!selectedRecipe.id) {
      toHome();
    }
  }, [selectedRecipe, toHome]);

  console.log(selectedRecipe);
  return (
    <Jumbotron>
      <Breadcrumb>
        <BreadcrumbItem>
          <a href="/home">Home</a>
        </BreadcrumbItem>
        <BreadcrumbItem active>Details</BreadcrumbItem>
      </Breadcrumb>
      {selectedRecipe.id && (
        <Row>
          <Col md={5}>
            <img
              src={selectedRecipe.image}
              alt={selectedRecipe.title}
              width="100%"
            />
            <Col md={12}>
              {selectedRecipe.dishTypes.map((dishType) => (
                <h3
                  key={`detail_card__footer-${selectedRecipe.id}-${dishType}`}
                >
                  <Badge color="success" outline="true">
                    {dishType}
                  </Badge>
                </h3>
              ))}
            </Col>
            <div
              dangerouslySetInnerHTML={{
                __html: selectedRecipe.summary,
              }}
            ></div>
          </Col>
          <Col md={7}>
            <h2>{selectedRecipe.title}</h2>
            <h3>Ingredients:</h3>
            <ul className="list-group">
              {selectedRecipe.extendedIngredients.map((ingredient, id) => (
                <li
                  className="list-group-item d-flex justify-content-between align-items-center"
                  key={`ingredient-${ingredient.name}-${id}`}
                >
                  {ingredient.name}
                  <span className="badge badge-info badge-pill">
                    {ingredient.amount} {ingredient.unit}
                  </span>
                </li>
              ))}
            </ul>
            <h3>Recipe link:</h3>
            <a href={selectedRecipe.spoonacularSourceUrl} target="_black">
              <img
                src="https://img.icons8.com/doodle/48/000000/cooking-book.png"
                alt="recipe icon"
              />
              Open recipe from Spoonacular!
            </a>
          </Col>
        </Row>
      )}
    </Jumbotron>
  );
}

const mapStateToProps = (state) => ({
  selectedRecipe: state.selectedRecipe,
});

const mapDispatchToProps = {};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Recipe));
