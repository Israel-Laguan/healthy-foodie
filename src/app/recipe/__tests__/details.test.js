import React from 'react';
import renderer from 'react-test-renderer';
import { Recipe } from '../detail';

describe('src/app/recipe:details', () => {
  it('render', () => {
    const props = {
      selectedRecipe: {
        id: '1',
        image: '#',
        title: 'My recipe',
        dishTypes: ['Italian'],
        summary: 'Some description',
        spoonacularSourceUrl: '#',
        extendedIngredients: [''],
      },
      history: {
        push: a => a,
      },
    };
    const component = renderer.create(
      <Recipe history={props.history} selectedRecipe={props.selectedRecipe} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
