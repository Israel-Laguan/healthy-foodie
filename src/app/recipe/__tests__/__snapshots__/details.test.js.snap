// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/recipe:details render 1`] = `
<div
  className="jumbotron"
>
  <nav
    aria-label="breadcrumb"
    className=""
  >
    <ol
      className="breadcrumb"
    >
      <li
        className="breadcrumb-item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        aria-current="page"
        className="active breadcrumb-item"
      >
        Details
      </li>
    </ol>
  </nav>
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <img
        alt="My recipe"
        src="#"
        width="100%"
      />
      <div
        className="col-md-12"
      >
        <h3>
          <span
            className="badge badge-success"
            outline="true"
          >
            Italian
          </span>
        </h3>
      </div>
      <div>
        Some description
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <h2>
        My recipe
      </h2>
      <h3>
        Ingredients:
      </h3>
      <ul
        className="list-group"
      >
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          <span
            className="badge badge-info badge-pill"
          >
             
          </span>
        </li>
      </ul>
      <h3>
        Recipe link:
      </h3>
      <a
        href="#"
        target="_black"
      >
        <img
          alt="recipe icon"
          src="https://img.icons8.com/doodle/48/000000/cooking-book.png"
        />
        Open recipe from Spoonacular!
      </a>
    </div>
  </div>
</div>
`;
